### ISSUE TITLE
[Bug]: react/jsx-key rule doesn't detect missing keys in LogicalExpression in return statements

### ISSUE THREAD

**Issue by yalperg** [jsx-eslint/eslint-plugin-react#3931]
### Is there an existing issue for this?

- [x] I have searched the existing issues and my issue is unique
- [x] My issue appears in the command-line and not only in the text editor

### Description Overview

The `react/jsx-key` rule fails to detect missing `key` props when JSX elements are conditionally rendered using the logical operators within return statements.

## Code example

```javascript
const list = [1, 2, 3, 4, 5, 6];

function Item({ item }) {
  return <p>{item}</p>;
}

function App() {
  return (
    <div>
      {list.map((item) => {
        return item < 4 && <Item item={item} />; // Missing key prop not detected
      })}
    </div>
  );
}

export default App;
```

### Expected Behavior

The rule should detect and report the missing `key` prop error.

### eslint-plugin-react version

v7.37.5

### eslint version

v9.28.0

### node version

v20.19.1

**Comment by toygma** [comment]
Hi! 👋
This is a valid issue and I’d be happy to help if you’re open to community contributions.
Please feel free to assign it to me if no one is working on it yet.

Thanks!
